src/main/java/com/keniu/mappers/UserMapper.java

```diff
@@ -29,6 +29,7 @@ public interface UserMapper {
      * @return the newly created User entity
      */
     @Mapping(target = "id", ignore = true)
+    @Mapping(target = "roles", ignore = true)
     @Mapping(target = "authorities", ignore = true)
     @Mapping(target = "isDeleted", expression = "java(false)")
```

Comment:
Your DTO doesn't consist these fields, so they won't be set to the User entity
```suggestion
```

Reply:
I see. This is way to tell MapStruct 'I haven't forgotten these fields' and to avoid getting warnings during application startup